src/rasterstats/main.py

```diff
@@ -205,14 +207,10 @@ def gen_zonal_stats(
                 if 'count' in stats:  # special case, zero makes sense here
                     feature_stats['count'] = 0
             else:
+                valid = masked.compressed()
+                keys, counts = np.unique(valid, return_counts=True)
                 if run_count:
-                    keys, counts = np.unique(masked.compressed(), return_counts=True)
-                    try:
-                        pixel_count = dict(zip([k.item() for k in keys],
-                                               [c.item() for c in counts]))
-                    except AttributeError:
-                        pixel_count = dict(zip([np.asscalar(k) for k in keys],
-                                               [np.asscalar(c) for c in counts]))
+                    pixel_count = dict(zip(keys.tolist(), counts.tolist()))
```

Comment:
`pixel_count` should really be a list of tuples. Floats do not make good keys in hash table.

Reply:
Good idea on the pixel_count calcs.

Tuples would work, though I'm hestitant to change that now due to backwards compatibility with existing code. I'd consider it thought. But In practice, categorical rasters (the only rasters it would make sense to request pixel counts on) are going to be some integer dtype. A categorical float would be exceedingly rare.

Reply:
Perhaps adding a note about this stat being intended for categorical rasters in the docstring would be enough.